test/functional/apm_logging_tests.js

```diff
@@ -0,0 +1,217 @@
+'use strict';
+
+const fs = require('fs');
+const stream = require('stream');
+const setupDatabase = require('./shared').setupDatabase;
+const chai = require('chai');
+const expect = chai.expect;
+const MongoClient = require('../..').MongoClient;
+const EJSON = require('mongodb-extjson');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+chai.use(sinonChai);
+
+const loggingEvents = [
+  'commandStarted',
+  'commandSucceeded',
+  'commandFailed',
+  'serverOpening',
+  'serverDescriptionChanged',
+  'serverHeartbeatStarted',
+  'serverHeartbeatSucceeded',
+  'serverHeartbeatFailed',
+  'serverClosed',
+  'topologyOpening',
+  'topologyClosed',
+  'topologyDescriptionChanged'
+];
+
+function stringify(obj) {
+  return EJSON.stringify(obj, { relaxed: true });
+}
+
+function getLogData(logArray) {
+  return logArray.reduce(
+    (data, event) => {
+      data.eventStrings.push(stringify(event.object));
+      data[event.category].push(event);
+      data.numLines += 1;
+      return data;
+    },
+    { numLines: 0, command: [], sdam: [], eventStrings: [] }
+  );
+}
+
+function runTestClient(context, urlOptions, closeCb) {
+  const configuration = context.configuration;
+  const url = configuration.url() + urlOptions;
+  const client = new MongoClient(url, { w: 1, useNewUrlParser: true });
+  const events = [];
+
+  loggingEvents.forEach(event => client.on(event, e => events.push(stringify(e))));
+
+  client.connect(function(err) {
```

Comment:
could you use arrow notation here since you're using it in `client.close` right below?